### PR TITLE
fixes #2690 - undefined method gsub! for nil:NilClass on fact import

### PR DESCRIPTION
**Pull request by isratrade**

### CODE REVIEW

File: app/models/facts_importer.rb [hunk 1]
```diff
@@ -30,8 +30,8 @@ def operatingsystem
           end
         end
         major, minor = orel.split(".")
-        major.gsub!(/\D/,'') unless is_numeric? major
-        minor.gsub!(/\D/,'') unless is_numeric? minor
+        major.to_s.gsub!(/\D/,'') unless is_numeric? major
+        minor.to_s.gsub!(/\D/,'') unless is_numeric? minor
         minor        ||= ""
```
Comment (domcleal) [inline app/models/facts_importer.rb]: Can this line be deleted, or perhaps it should be moved above instead of using to_s?

**Comment by domcleal** [comment]
Would you mind adding a test or two around this?  Thanks!

**Comment by isratrade** [comment]
@domcleal, re-committed

**Comment by domcleal** [comment]
:+1: great, thanks @isratrade.  I'll wait for tests.

**Comment by domcleal** [comment]
@isratrade failures I'm afraid on all databases, e.g.

<pre>
Mysql2::Error: Column 'minor' cannot be null: INSERT INTO `operatingsystems` (`created_at`, `major`, `minor`, `name`, `nameindicator`, `release_name`, `type`, `updated_at`) VALUES ('2013-06-24 11:07:23', '6', NULL, 'AnyOS', NULL, NULL, NULL, '2013-06-24 11:07:23')
</pre>

**Comment by isratrade** [comment]
@domcleal, the on the database layer for minor is 
not null default ''::character varying

I will check and re-submit

**Comment by isratrade** [comment]
@domcleal, re-submitted.  I moved .to_s to a different place.

**Comment by isratrade** [comment]
@domcleal, re-submitted again.  I need .to_s in the args { }.  Always a good idea to run tests locally, right :-)

**Comment by domcleal** [comment]
Merged as 9628014.  Thanks @isratrade!

tests++ :)

**Pull request closed without merging**